src/KiD_driver.jl

```diff
@@ -23,59 +27,46 @@ n_elem = 256
 t_ini = 0.0
 t_end = 10.0 * 60
 
-# Updraft momentum flux terms and initial conditions
+# Updraft momentum flux terms, surface terms and initial conditions
 w1 = 2 # m/s * kg/m3
 t1 = 600 # s
 w_params = (w1 = w1, t1 = t1)
+q_surf = 0.016
+ρw0 = 0.0
 
-domain = Domains.IntervalDomain(
-    Geometry.ZPoint{FT}(z_min),
-    Geometry.ZPoint{FT}(z_max),
-    boundary_names = (:bottom, :top),
-)
-mesh = Meshes.IntervalMesh(domain, nelems = n_elem)
+# initialize the timestepping struct
+TS = TimeStepping(FT(Δt), FT(Δt_output), FT(t_end))
+
+# create the coordinates,
+space, face_space = make_function_space(z_min, z_max, n_elem)
 
-space = Spaces.CenterFiniteDifferenceSpace(mesh)
-face_space = Spaces.FaceFiniteDifferenceSpace(space)
+coord = CC.Fields.coordinate_field(space)
+face_coord = CC.Fields.coordinate_field(face_space)
 
-coord = Fields.coordinate_field(space)
-face_coord = Fields.coordinate_field(face_space)
+# initialize the netcdf output Stats struct
+Stats = NetCDFIO_Stats("Output.nc", 1.0, vec(face_coord), vec(coord))
 
 # solve the initial value problem for density profile
 ρ_profile = ρ_ivp(FT, params)
 # create the initial condition profiles
 init = map(coord -> init_1d_column(FT, params, ρ_profile, coord.z), coord)
-ρw = Geometry.WVector.(zeros(FT, face_space))
-ρw0 = 0.0
 
-# initialize the netcdf output Stats struct
-Stats = NetCDFIO_Stats("Output.nc", 1.0, vec(face_coord), vec(coord))
-# initialize the timestepping struct
-TS = TimeStepping(FT(Δt), FT(Δt_output), FT(t_end))
+# create state vector and apply initial condition
+Y = initialise_state(EquilibriumMoisture(), NoPrecipitation(), init)
```

Comment:
One good thing could be to make this into a test file, and then write a test file for each case of moisture, precip, etc.

Reply:
Opened an issue #21